### DISCUSSION TITLE
Use the new `@vercel/postgres` package with supabase?

### DISCUSSION THREAD

**Discussion by julianpoma** [supabase/supabase#14165] [Questions]
Has anyone tried the combo in the title?

What I like about the new `@vercel/postgres` over `node-postgres` one is the `sql` function to write my queries using string templates. 

I am getting the following error:

`Error [VercelPostgresError]: VercelPostgresError - 'invalid_connection_string': This connection string is meant to be used with a direct connection. Make sure to use a pooled connection string or try createClient() instead.`

I am using the "pooled" URL that Supabase provides!

**Comment by jschuur** [comment]
> I am using the "pooled" URL that Supabase provides!

Their pooled URL check seems to be [hardcoded](https://github.com/vercel/storage/blob/8b51d484e3a0b3e3f0f487e7e6313b5a8274dd50/packages/postgres/src/postgres-connection-string.ts#L33) for the presence of `-pooler.` in it right now.

see https://github.com/vercel/storage/issues/123

**Comment by remorses** [comment]
You can currently fix this with

```ts

const uri = DATABASE_URL + '?-pooler.',

```

**Reply by seivad** [reply]
It's crazy that this is the fix. I am using Supabase and for example my connection string has ```aws-0-us-west-1.pooler.supabase.com``` within it but the .includes() does not pick up pooler. because it's looking for "dash" "pooler.". Can the guys Vercel please change the .includes to just be .includes('pooler') and then we won't need this hack at the end of the url string.

**Reply by e4stwood** [reply]
+1 - This fixed my use of vercel/platforms with a custom db. As above, _insane_ that this is how I have to hack around Vercel

**Comment by zeloxx** [comment]
try this:

```ts
import { drizzle } from 'drizzle-orm/vercel-postgres'
import { createPool } from '@vercel/postgres'

const client = createPool({
    //! WHAT IN THE ACTUAL FUCK VERCEL
    connectionString: process.env.DATABASE_URL + '?workaround=supabase-pooler.vercel',
})

export const db = drizzle(client, {
    schema: dbSchema,
})
```

**Comment by qbaze** [comment]
one thing to add 

for me the workaround works for getting rid of the `invalid_connection_string` exception

but ...

a new problem pops up, once `sql()` is called on the pool

it tries to connect to `api.pooler.supabase.com` instead of the originally provided db url `aws-0-eu-central-1.pooler.supabase.com` and fails as it does not exist

why?

the `vercel/postgres` is built on top of the `neon/serverless` library

its `sql()` (vercel lib) method uses the `neon()` (neon lib) function in its body [here](https://github.com/vercel/storage/blob/dca9772d45c8403abf7d8fadaea97310666f9b8e/packages/postgres/src/create-pool.ts#L46) and the connection string is then being transformed from postgresql protocol to http representation [here](https://github.com/neondatabase/serverless/blob/c944e698e4dfe4c25b522b434fe57d56352b9923/shims/net/index.ts#L66)

`query()` function does not use `neon()` for the conn string and works as expected

imho there are too many undocumented workarounds needed that make this lib useless for supabase or anything else then vercel db and this should be clearly stated in the documentation

**Reply by hallboav** [reply]
Thanks for you answer @qbaze
What should I do in order to have the sql`SELECT 1` working?
I have a POSTGRES_URL set and I always face a error trying to connect to api.pooler.supabase.com (this is not the correct host as you stated)

**Reply by qbaze** [reply]
hey @hallboav

1. create a pool 
`const pool = createPool({
    connectionString: process.env.POSTGRES_URL,
  });
`
2. use `query` instead of `sql` method for your queries
`await pool().query('SELECT 1')`
